src/Tasks/Microsoft.NET.Build.Tasks/GetPackagesToPrune.cs

```diff
@@ -72,51 +83,219 @@ protected override void ExecuteCore()
+                    //  we can load it from the targeting packs that are packaged with the SDK.
+                    if (packagesForFrameworkReference == null)
+                    {
+                        log.LogMessage("Failed to load prune package data from PrunePackageData folder, loading from targeting packs instead");
+                        packagesForFrameworkReference = LoadPackagesToPruneFromTargetingPack(log, key.TargetFrameworkIdentifier, key.TargetFrameworkVersion, frameworkReference, targetingPackRoots);
+                    }
+
+                    //  Fall back to framework packages data for older framework for WindowsDesktop if necessary
+                    //  https://github.com/dotnet/windowsdesktop/issues/4904
+                    if (packagesForFrameworkReference == null && frameworkReference.Equals("Microsoft.WindowsDesktop.App", StringComparison.OrdinalIgnoreCase))
+                    {
+                        log.LogMessage("Failed to load prune package data for WindowsDesktop from targeting packs, loading from framework packages instead");
+                        packagesForFrameworkReference = LoadPackagesToPruneFromFrameworkPackages(key.TargetFrameworkIdentifier, key.TargetFrameworkVersion, frameworkReference,
+                            acceptNearestMatch: true);
+                    }
+                }
+
+                if (packagesForFrameworkReference == null)
+                {
+                    //  We didn't find the data for packages to prune.  This indicates that there's a bug in the SDK construction, so fail hard here so that we fix that
+                    //  (rather than a warning that might be missed).
+                    //  Since this indicates an error in the SDK build, the message probably doesn't need to be localized.
+
+                    if (allowMissingPrunePackageData)
+                    {
+                        log.LogMessage($"Prune package data not found for {key.TargetFrameworkIdentifier} {key.TargetFrameworkVersion} {frameworkReference}");
+                    }
+                    else
+                    {
+                        log.LogError(Strings.PrunePackageDataNotFound, key.TargetFrameworkIdentifier, key.TargetFrameworkVersion, frameworkReference);
+                    }
+                }
+                else
+                {
+                    AddPackagesToPrune(packagesToPrune, packagesForFrameworkReference.Select(kvp => (kvp.Key, kvp.Value)), log);
                 }
             }
 
-            PackagesToPrune = packagesToPrune.Select(p =>
+
+
+            return packagesToPrune.Select(p =>
             {
                 var item = new TaskItem(p.Key);
-                item.SetMetadata("Version", p.Value.ToString());
+
+                string version;
+                if (key.TargetFrameworkIdentifier.Equals(".NETCoreApp", StringComparison.OrdinalIgnoreCase))
```

Comment:
Was having some discussion with @ViktorHofer about this and it might also be better to avoid this lifting when `!p.Value.IsPrerelease`.   What are your thoughts on this @dsplaisted?
```suggestion
                if (key.TargetFrameworkIdentifier.Equals(".NETCoreApp", StringComparison.OrdinalIgnoreCase) && !p.Value.IsPrerelease)
```